### DISCUSSION TITLE
ScriptReference/MonoBehaviour.OnCollisionEnter

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#490] [Notes]
# ScriptReference/MonoBehaviour.OnCollisionEnter

https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnCollisionEnter.html

**Comment by jschiff** [comment]
Note that collision detection runs as part of the physics simulation phase of the game loop, which leads to a few gotchas. For example, OnCollisionEnter will _not_ run in any of the following circumstances:
- [Time.timeScale](https://docs.unity3d.com/ScriptReference/Time-timeScale.html) is set to 0
- [Physics.autoSimulation](https://docs.unity3d.com/ScriptReference/Physics-autoSimulation.html) is disabled (via script or in project settings)
- Objects are moved into an overlapping state manually at any point in the game loop, e.g. Update, or even FixedUpdate. The collision will not happen until the physics simulation runs.
